jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/ApplyService.java

```diff
@@ -1390,4 +1394,55 @@ public void processCustomEntities(List<String> customResourceDefinitions, File r
             }
         }
     }
+
+    public void applyNamespaceOrProjectIfPresent(Collection<HasMetadata> entities) {
+        List<HasMetadata> projectOrNamespaceList = entities.stream()
+                .filter(e -> e instanceof Namespace || e instanceof Project)
+                .collect(Collectors.toList());
+
+        for (HasMetadata projectOrNamespace : projectOrNamespaceList) {
```

Comment:
why not do a `forEach` here.
You are generating a list of HasMetadata, and then re-iterating over that list.